README.md

```diff
@@ -1,196 +1,289 @@
+  - `workflow`: (Optional) ID of a DNAnexus workflow to be triggered after successful upload of the RUN directory. This workflow's I/O contract should accept a DNAnexus record with the  name `upload_sentinel_record` in the 1st stage (stage 0) of the workflow as input. Additional input can be specified using the variable `downstream_input`. **Note that if the specified workflow is not located, the upload process will not commence. Mutually exclusive with `applet`. The role will raise an error and fail if both are specified.**
+  - `downstream_input`: (Optional) A JSON string, parsable as a python `dict` of `str`:``str`, where the **key** is the input_name recognized by a DNAnexus applet/workflow and the **value** is the corresponding input. For examples and detailed explanation, see section titled `Downstream analysis`. **Note that the role will raise an error and fail if this string is not JSON-parsable as a dict of the expected format**
+**Note** DNAnexus login is persistent and the login environment is stored on disk in the the Ansible user's home directory. User of this playbook responsibility to make sure that every Ansible user (`monitored_user`) with a streaming upload job assigned has been logged into DNAnexus by either specifying a `dx_token` or `dx_user_token`.
+#### Example Playbook
+`dx-upload-play.yml`
+```YAML
+---
+- hosts: localhost
+  vars:
+    monitored_users:
+      - username: travis
+        local_tar_directory: ~/new_location/upload/TMP
+        local_log_directory: ~/another_location/upload/LOG
+        monitored_directories:
+          - ~/runs
+        applet: applet-Bq2Kkgj08FqbjV3J8xJ0K3gG
+        downstream_input: '{"sequencing_center": "CENTER_A"}'
+      - username: root
+        monitored_directories:
+          - ~/home/root/runs
+        workflow: workflow-BvFz31j0Y7V5QPf09x9y91pF
+        downstream_input: '{"0.sequencing_center: "CENTER_A"}'
+    mode: debug
+    upload_project: project-BpyQyjj0Y7V0Gbg7g52Pqf8q
+  roles:
+    - dx-streaming-upload
+```
+**Note**: For security reasons, you should refrain from storing the DNAnexus authentication token in a playbook that is open-access. One might trigger the playbook on the command line with extra-vars to supply the necessary authentication token, or store them in a closed-source yaml variable file.
+ie. `ansible-playbook dx-upload-play.yml -i inventory --extra-vars "dx_token=<SECRET_TOKEN>"`
+We recommend that the token given is limited in scope to the upload project, and has no higher than **CONTRIBUTE** privileges.

```

Comment:
--> " We recommend that the token given is limited in scope to the upload project and has no higher than **CONTRIBUTE** privileges. "